frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,368 @@
+/* eslint-disable no-undef */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+
+import { connectToPlural } from '../../kinds';
+import { k8sCreate, K8sKind, referenceFor } from '../../module/k8s';
+import { ButtonBar, history, LoadingBox, NameValueEditor, resourceObjPath, RequestSizeInput } from '../utils';
+import { ListDropdown } from '../RBAC/bindings';
+import { RadioInput } from '../radio';
+import { Checkbox } from '../checkbox';
+
+const StorageClassDropdown: React.SFC<StorageClassDropdownProps> = props => {
+  const kind = 'StorageClass';
+  const resources = [{ kind }];
+  const { selectedKey, required, namespace } = props;
+  return (
+    <ListDropdown
+      {...props}
+      desc="Storage Classes"
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select storage class"
+      selectedKey={selectedKey}
+      required={required}
+      namespace={namespace}
+      name={name}
+    />
+  );
+};
+// This form is done a little odd since it is used in both its own page and as
+// a sub form inside the attach storage page.
+export class CreatePvcForm extends React.Component<CreatePvcFormProps, CreatePvcFormState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      storageClass: '',
+      pvcName: '',
+      accessMode: 'ReadWriteOnce',
+      requestSizeValue: '1',
+      requestSizeUnit: 'Gi',
+      disableForm: false,
+      showLabelSelectors: false,
+      nameValuePairs: [['', '']],
+      radios: [
+        {
+          value: 'ReadWriteOnce',
+          title: 'Single User (RWO)',
+        },
+        {
+          value: 'ReadWriteMany',
+          title: 'Shared Access (RWX)',
+        },
+        {
+          value: 'ReadOnlyMany',
+          title: 'Read Only (ROX)',
+        },
+      ],
+      dropdownUnits: {
+        Mi: 'MiB',
+        Gi: 'GiB',
+        Ti: 'TiB',
+      },
+    };
+  }
+
+  handleChange: React.ReactEventHandler<HTMLInputElement> = event => {
+    // this handles pvcName, accessMode, size
+    const { name, value } = event.currentTarget;
+    this.setState({ [name]: value } as any, () => {
+      this.onChange();
+    });
+  };
+
+  handleNameValuePairs = ({ nameValuePairs }) => {
+    this.setState({ nameValuePairs }, () => {
+      this.onChange();
+    });
+  };
+
+  handleStorageClass = storageClass => {
+    this.setState({ storageClass }, () => {
+      this.onChange();
+    });
+  };
+
+  handleClear = () => {
```

Comment:
This needs a more specific name, otherwise it sounds like we're clearing the entire form. Maybe `clearStorageClass`.